### PR TITLE
Add Eigen 3.3.5

### PR DESCRIPTION
**Pull request by norbertwenzel**
Use the (afaik) official Eigen Git mirror as source now.
Make Eigen 3.3.5 the new default version.

I did not test this changes yet. I only compared the latest Hunter Eigen 3.3.4-p1 archive with Eigen's own 3.3.4 archive, and the only difference I managed to find where differing hashes in `./pax_global_header`. So the main question to me currently is, whether changing the Eigen archive to eigen-git-mirror is welcome?

### CODE REVIEW

**Comment by NeroBurner** [comment]
deep-pkg-testing updated, some old toolchains deactivated, some new ones added. Please review
CI: 
- https://ci.appveyor.com/project/NeroBurner/hunter/build/1.0.70
- https://travis-ci.org/NeroBurner/hunter

yml-files:
- travis file: https://github.com/NeroBurner/hunter/blob/77cd7f5536de0815c7bef294cc7c311e6b970c5e/.travis.yml
- appveyor file: https://github.com/NeroBurner/hunter/blob/77cd7f5536de0815c7bef294cc7c311e6b970c5e/appveyor.yml

File: cmake/configs/default.cmake [hunk 1]
```diff
@@ -58,7 +58,7 @@ hunter_default_version(Comet VERSION 4.0.2)
 hunter_default_version(CppNetlib VERSION 0.10.1-hunter-3)
 hunter_default_version(CppNetlibUri VERSION 1.0.5-hunter)
 hunter_default_version(CsvParserCPlusPlus VERSION 1.0.1)
-hunter_default_version(Eigen VERSION 3.3.4-p1)
+hunter_default_version(Eigen VERSION 3.3.5-p0)
```
Comment (NeroBurner) [inline cmake/configs/default.cmake]: this should be version `3.3.5`. The `-p0` extension indicates a hunter specific patch

Comment (norbertwenzel) [reply cmake/configs/default.cmake]: @NeroBurner fixed in 1685d34

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.4346
* https://travis-ci.org/ingenue/hunter/builds/420510508?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Uploading:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.4348
* https://travis-ci.org/ingenue/hunter/builds/420519834?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Released: https://github.com/ruslo/hunter/releases/tag/v0.23.17

**Pull request merged**